src/log/setup.cr

```diff
@@ -0,0 +1,39 @@
+class Log
+  # Setups logging bindings discarding all previous configurations.
+  def self.setup(*, builder : Log::Builder = Log.builder)
+    builder.clear
+    yield builder.as(Configuration)
+  end
+
+  # Setups logging for all sources using the specified *level*, *backend*.
+  def self.setup(level : Log::Severity = Log::Severity::Info,
+                 backend : Log::Backend = IOBackend.new,
+                 *, builder : Log::Builder = Log.builder)
+    Log.setup("*", level, backend, builder: builder)
+  end
+
+  # Setups logging for *sources* using the specified *level*, *backend*.
+  def self.setup(sources : String = "*",
```

Comment:
I'd rather get rid of all the sources-based `Log.setup` overloads in this PR, and leave it until the infra for parsing `k=v` is merged. This is because we'll then have a canonical way of easily specifying log levels, and the `Log.setup` interface here behaves quite differently (for example not keeping the *=INFO default, instead disabling logging for all but the mentioned sources), creating confusion.

I'll accept it if you want to leave these in, but to me, I think people will be fine with the `Log.setup(&block)` interface *for now* if they want to setup per-source levels.

Reply:
I think those overloads are useful for simple scenarios, where even configuring through environment variables is not needed or undesired. For example `Log.setup(:warning)` or `Log.setup("http.server", :info)` to just show request logs. Do you say you would use the same syntax as environment variables instead, even for programmatic setup?

It has to be understood that calling `Log.setup` is never additive. It always replace the current configuration.

Reply:
> It has to be understood that calling `Log.setup` is never additive. It always replace the current configuration.

I'm just not sure that this is the semantic that users want in practice. For all means, I'm not going to delay this PR over that discussion though.